src/services/ghost/GhostInlineCompletionProvider.ts

```diff
@@ -0,0 +1,869 @@
+	 */
+	private isPrefix(deletedContent: string, addedContent: string): boolean {
+		return addedContent.startsWith(deletedContent)
+	}
+
+	/**
+	 * Check if deletion+addition should be treated as pure addition
+	 */
+	private shouldTreatAsAddition(deletedContent: string, addedContent: string): boolean {
+		// If added content starts with deleted content, let common prefix logic handle this
+		if (this.isPrefix(deletedContent, addedContent)) {
+			return false
+		}
+
+		// Added content starts with newline - indicates LLM wants to add content after current line
+		return addedContent.startsWith("\n") || addedContent.startsWith("\r\n")
+	}
+
+	private shouldShowGroup(groupType: "+" | "/" | "-", group?: GhostSuggestionEditOperation[]): boolean {
+		// Always show pure additions
+		if (groupType === "+") {
+			return true
+		}
+
+		// For modifications, allow completions with common prefix
+		if (groupType === "/" && group) {
+			const deletedContent = this.extractContent(group, "-")
+			const addedContent = this.extractContent(group, "+")
+
+			if (deletedContent && addedContent && this.isPrefix(deletedContent, addedContent)) {
+				return true
+			}
+		}
+
+		// Don't show deletions or non-prefix modifications
+		return false
+	}
+
+	/**
+	 * Calculate target line for a group
+	 */
+	private getTargetLine(
+		group: GhostSuggestionEditOperation[],
+		groupType: "+" | "/" | "-",
+		offset: { added: number; removed: number },
+	): number {
+		// For modifications, use the deletion line without offsets
+		if (groupType === "/") {
+			const deleteOp = group.find((op) => op.type === "-")
+			return deleteOp ? deleteOp.line : group[0].line
+		}
+
+		// For additions, apply the offset to account for previously removed lines
+		if (groupType === "+") {
+			return group[0].line + offset.removed
+		}
+
+		// For deletions
+		return group[0].line + offset.added
+	}
+
+	/**
+	 * Check if group is within cursor distance
+	 */
+	private isWithinCursorDistance(cursorLine: number, targetLine: number): boolean {
+		return Math.abs(cursorLine - targetLine) <= MAX_CURSOR_DISTANCE
+	}
+
+	/**
+	 * Check if modification has valid common prefix for inline completion
+	 */
+	private hasValidCommonPrefixForInline(group: GhostSuggestionEditOperation[]): boolean {
+		const deletedContent = this.extractContent(group, "-")
+		const addedContent = this.extractContent(group, "+")
+
+		if (!deletedContent || !addedContent) {
+			return false
+		}
+
+		// If deleted content is empty or placeholder, treat as pure addition
+		const trimmedDeleted = deletedContent.trim()
+		if (trimmedDeleted.length === 0 || this.isPlaceholderContent(trimmedDeleted)) {
+			return true
+		}
+
+		// Check if should be treated as addition
+		if (this.shouldTreatAsAddition(deletedContent, addedContent)) {
+			return true
+		}
+
+		// Check for common prefix
+		const commonPrefix = this.findCommonPrefix(deletedContent, addedContent)
+		return commonPrefix.length > 0
```

Comment:
is this enough? Just the addition and deletion starting with a space because each line does? This feels like a bug tbh